tests/test_command_parse.py

```diff
@@ -70,6 +70,19 @@ class MyBadCrawlSpider(CrawlSpider):
 
     def parse(self, response):
         return [scrapy.Item(), dict(foo='bar')]
+
+
+class MyInfiniteCallbackSpider(scrapy.Spider):
+
+    name = 'infinitecb{0}'
+
+    def parse(self, response):
+        if response.meta.get('retried'):
+            yield dict(foo='bar')
+            return
+
+        response.meta['retried'] = True
+        yield response.request.replace(dont_filter=True)
```

Comment:
May you can add another test case that tests non-default callbacks, like the example [in this comment](https://github.com/scrapy/scrapy/pull/3124#discussion_r169702883)?